extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js

```diff
@@ -41,28 +48,71 @@ export const URL_EXPERIMENT_MAPPING = {
   '0': null,
 };
 
+export const ADSENSE_EXPERIMENTS = {
+  UNCONDITIONED_CANONICAL_EXP: '21062154',
+  UNCONDITIONED_CANONICAL_CTL: '21062155',
+  CANONICAL_EXP: '21062158',
+  CANONICAL_CTL: '21062159',
+};
+
+export const ADSENSE_EXP_NAMES = {
+  UNCONDITIONED_CANONICAL: 'expAdsenseUnconditionedCanonical',
+  CANONICAL: 'expAdsenseCanonical',
+};
+
 /**
+ * Attempts to select into Adsense experiments.
  * @param {!Window} win
  * @param {!Element} element
- * @param {boolean} useRemoteHtml
- * @return {boolean}
  */
-export function adsenseIsA4AEnabled(win, element, useRemoteHtml) {
-  if (useRemoteHtml || !isGoogleAdsA4AValidEnvironment(win) ||
-      !element.getAttribute('data-ad-client')) {
-    return false;
-  }
+function selectExperiments(win, element) {
+  selectAndSetExperiments(win, element,
+      [ADSENSE_EXPERIMENTS.UNCONDITIONED_CANONICAL_EXP,
+        ADSENSE_EXPERIMENTS.UNCONDITIONED_CANONICAL_CTL],
+      ADSENSE_EXP_NAMES.UNCONDITIONED_CANONICAL,
+      true);
+
+
   // See if in holdback control/experiment.
   let experimentId;
   const urlExperimentId = extractUrlExperimentId(win, element);
   if (urlExperimentId != undefined) {
     experimentId = URL_EXPERIMENT_MAPPING[urlExperimentId];
-    dev().info(
-        TAG, `url experiment selection ${urlExperimentId}: ${experimentId}.`);
   }
   if (experimentId) {
     addExperimentIdToElement(experimentId, element);
     forceExperimentBranch(win, ADSENSE_A4A_EXPERIMENT_NAME, experimentId);
+    dev().info(
+        TAG, `url experiment selection ${urlExperimentId}: ${experimentId}.`);
+  }
+
+  // If not in the unconditioned canonical experiment, attempt to
+  // select into the undiluted canonical experiment.
+  const inUnconditionedCanonicalExp = !!getExperimentBranch(
+      win, ADSENSE_EXP_NAMES.UNCONDITIONED_CANONICAL);
+  if (!inUnconditionedCanonicalExp && isCanonical(win)) {
+    selectAndSetExperiments(win, element,
+        [ADSENSE_EXPERIMENTS.CANONICAL_EXP,
+          ADSENSE_EXPERIMENTS.CANONICAL_CTL],
+        ADSENSE_EXP_NAMES.CANONICAL, true);
+  }
+}
+
+/**
+ * @param {!Window} win
+ * @param {!Element} element
+ * @param {boolean} useRemoteHtml
+ * @return {boolean}
+ */
+export function adsenseIsA4AEnabled(win, element, useRemoteHtml) {
+  if (useRemoteHtml || !element.getAttribute('data-ad-client')) {
+    return false;
   }
-  return true;
+  selectExperiments(win, element);
+  return isGoogleAdsA4AValidEnvironment(win) ||
```

Comment:
What if I'm in the manual experiment (#exp=a4a:-1)?  That should force fast fetch

Reply:
discussed offline